p2p/host/autorelay/autorelay.go

```diff
@@ -135,6 +154,65 @@ func (ar *AutoRelay) background(ctx context.Context) {
 	}
 }
 
+func (ar *AutoRelay) refreshReservations(ctx context.Context, now time.Time) bool {
+	ar.mx.Lock()
+	if ar.status == network.ReachabilityPublic {
+		// we are public, forget about the relays, unprotect peers
+		for p := range ar.relays {
+			ar.host.ConnManager().Unprotect(p, autorelayTag)
+			delete(ar.relays, p)
+		}
+
+		ar.mx.Unlock()
+		return true
+	}
+
+	// find reservations about to expire and refresh them in parallel
+	var (
+		wg   sync.WaitGroup
+		fail int32
+	)
+
+	for p, rsvp := range ar.relays {
+		if rsvp == nil {
+			continue
+		}
+
+		if now.Add(rsvpExpirationSlack).Before(rsvp.Expiration) {
+			continue
+		}
+
+		wg.Add(1)
+		go ar.refreshRelayReservation(ctx, p, &wg, &fail)
```

Comment:
This would be a good use case for the `errgroup` as in this example: https://pkg.go.dev/golang.org/x/sync/errgroup#example-Group-JustErrors.

Reply:
I thought about it (i like this package a lot too), but it is not necessarily an error if we fail to refresh a reservation -- maybe the relay got overloaded.

Reply:
having said that, i won't object if you think we should do it with an errgrp.

Reply:
actually on further thought, yeah, the code will be simpler and easier to grok, so I'll do the errgrp thing.

Reply:
done.